### PR TITLE
Mapped out tools and related things

### PR DESCRIPTION
**Pull request by Darkhax**
This maps out some of the methods and fields that are related to tool classes.

### CODE REVIEW

File: mappings/net/minecraft/item/tool/ItemTool.mapping [hunk 1]
```diff
@@ -1,5 +1,35 @@
...
+		ARG 1 state
+	METHOD a onEntityDamaged (Lnone/aeq;Lnone/sn;Lnone/sn;)Z
+		ARG 0 stack
+		ARG 1 target
+		ARG 2 user
+	METHOD a getAttributeModifiers (Lnone/sj;)Lcom/google/common/collect/Multimap;
+		ARG 0 slotType
+	METHOD c getEnchantability ()I
 	METHOD g getMaterial ()Lnone/aeo$a;
+	METHOD h getMaterial ()Ljava/lang/String;
```
Comment (asiekierka) [inline mappings/net/minecraft/item/tool/ItemTool.mapping]: Two methods with the same name which output different things? Looks like a bug.

File: mappings/net/minecraft/item/tool/ItemHoe.mapping [hunk 2]
```diff
@@ -1,2 +1,19 @@
 CLASS none/aen net/minecraft/item/tool/ItemHoe
+	FIELD a material Lnone/aeo$a;
+	FIELD b swingSpeed F
+	METHOD <init> (Lnone/aeo$a;)V
+		ARG 0 material
 	METHOD a activate (Lnone/aak;Lnone/aiu;Lnone/cn;Lnone/qz;Lnone/cu;FFF)Lnone/rb;
+	METHOD a tillSoil (Lnone/aeq;Lnone/aak;Lnone/aiu;Lnone/cn;Lnone/asj;)V
```
Comment (asiekierka) [inline mappings/net/minecraft/item/tool/ItemHoe.mapping:7]: Is there a simpler way to word this? Is it specific to the Hoe?

Comment (Darkhax) [inline mappings/net/minecraft/item/tool/ItemHoe.mapping:7]: Yes, this method is only used by the hoe, and is only in the hoe class. It's basically a fancy way for the how to set a block and play it's sound.

File: mappings/net/minecraft/item/tool/ItemTool.mapping (hunk 1 above)
Comment (Darkhax) [inline mappings/net/minecraft/item/tool/ItemTool.mapping]: This is a typo on my end. The one which returns String should be getMaterialName. Will fix.

File: mappings/net/minecraft/item/tool/ItemHoe.mapping (hunk 2 above)
Comment (shadowfacts) [inline mappings/net/minecraft/item/tool/ItemHoe.mapping:7]: I think `tillSoil` is fine, that's what the method does.

**Comment by shadowfacts** [comment]
LGTM

**Pull request merged**